### PR TITLE
[8.x] Add messageId method to Message class

### PR DESCRIPTION
**Pull request by u12206050**
Added helper function `messageId` to retrieve X-Message-ID function from the headers of the swift message.

### CODE REVIEW

File: src/Illuminate/Mail/Message.php [hunk 1]
```diff
@@ -315,6 +315,16 @@ public function getSwiftMessage()
         return $this->swift;
     }
 
+    /**
+     * @return string|int|null
```
Comment (driesvints) [inline src/Illuminate/Mail/Message.php:319]: Add a method description on top.

**Comment by taylorotwell** [comment]
Can you show me how you would use this in an application?

**Comment by u12206050** [comment]
This is how we currently use it (simplified) I realise this is a bit ugly. Would be much better if Message Events sent back `Illuminate\Mail\Message` instead of `Swift_Message`. Shall I make a pull request making that change for the next version?

```
// EventServiceProvider
/**
 * The event listener mappings for the application.
 *
 * @var array
 */
protected $listen = [
    'Illuminate\Mail\Events\MessageSent' => [
        'App\Listeners\TrackSentMessage',
    ],
];
```

```
// App\Listeners\TrackSentMessage

    public function handle(MessageSent $event)
    {
          $message = $event->data['message'];
          $message->messageId();

          // Save to Database or something else
    }
```

**Comment by u12206050** [comment]
Using `Illuminate\Mail\Message` as the message instead would look like this:

 ```
// App\Listeners\TrackSentMessage

    public function handle(MessageSent $event)
    {
          
          /* @var \Illuminate\Mail\Message $message */
          $message = $event->message;
          $message->messageId();

         // To get swift message (update guide)
         $swiftMessage = $message->getSwiftMessage();
    }
```

**Comment by taylorotwell** [comment]
For now just access it off the Swift message. Part of me would expect this method to set the message ID, not retrieve it.

**Pull request closed without merging**